### PR TITLE
Changes needed to make dotnet work with nsjail

### PR DESCRIPTION
**Pull request by partouf**
Changes on top of #3168 to make dotnet work on our website

### CODE REVIEW

File: etc/nsjail/execute.cfg [hunk 1]
```diff
@@ -10,7 +10,7 @@ log_level: FATAL
 rlimit_as_type: INF
 rlimit_cpu_type: SOFT
 rlimit_fsize: 1024 # 1GB
-rlimit_nofile: 100
+rlimit_nofile: 150
```
Comment (partouf) [inline etc/nsjail/execute.cfg:13]: this is based on a bit of experimentation, 125 was not enough, 150 was

File: lib/compilers/dotnet.ts [hunk 2]
```diff
@@ -26,13 +26,29 @@ import path from 'path';
 
 import fs from 'fs-extra';
 
+/// <reference types="../base-compiler" />
```
Comment (partouf) [inline lib/compilers/dotnet.ts:29]: this is the trick to enable typing on javascript imports

File: etc/config/csharp.amazon.properties [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+compiler.dotnet601csharp.exe=/opt/compiler-explorer/dotnet-v6.0.1/bin/coreclr/Linux.x64.Checked/.dotnet/dotnet
+compiler.dotnet601csharp.name=.NET 6.0.101
+compiler.dotnet601csharp.clrDir=/opt/compiler-explorer/dotnet-v6.0.1/bin/coreclr/Linux.x64.Checked
+compiler.dotnet601csharp.runtimeId=linux-x64
+compiler.dotnet601csharp.targetFramework=net6.0
+compiler.dotnet601csharp.buildConfig=Release
+compiler.dotnet601csharp.additionalSources=
+compiler.dotnet601csharp.langVersion=latest
+compiler.dotnet601csharp.nugetPackages=/opt/compiler-explorer/dotnet_nuget-v6.0.0/packages
```
Comment (partouf) [inline etc/config/csharp.amazon.properties:17]: this new path contains pre-downloaded nuget packages for dotnet core, aspnet core and fsharp core

**Comment by partouf** [comment]
well this is new, the build fails because `nyc npm run test` produces an error that `npm run test` on its own does not...

**Comment by partouf** [comment]
> well this is new, the build fails because `nyc npm run test` produces an error that `npm run test` on its own does not...

Fixed

File: etc/nsjail/execute.cfg (hunk 1 above)
Comment (mattgodbolt) [reply etc/nsjail/execute.cfg:13]: thanks... 100 was a bit made up before :)

**Review by mattgodbolt — APPROVED** [review]
Great!

**Comment by partouf** [comment]
now live

**Pull request merged**